crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_test_hooks.rs

```diff
@@ -0,0 +1,221 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, AddVisitor, Phases, QueryMatch, Queryable, Rule,
+    RuleDiagnostic, ServiceBag, Visitor, VisitorContext,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsCallExpression, JsLanguage, JsSyntaxToken, TextRange};
+use biome_rowan::{AstNode, Language, SyntaxNode, WalkEvent};
+
+declare_rule! {
+    /// A `describe` block should not contain duplicate hooks.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///    // some test
+    ///   });
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///     afterAll(() => {
+    ///       // some teardown
+    ///     });
+    ///    afterAll(() => {
+    ///      // some teardown
+    ///    });
+    ///   });
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///     // some test
+    ///   });
+    /// });
+    /// ```
+    ///
+    pub(crate) NoDuplicateTestHooks {
+        version: "next",
+        name: "noDuplicateTestHooks",
+        recommended: true,
```

Comment:
We should add the `source`. It should be `source: RuleSource::EslintJest`, and we should add `source_kind` as `RuleSourceKind::Inspired`